test/test-tools.go

```diff
@@ -178,3 +178,27 @@ func CountHistogramSamples(hist prometheus.Histogram) int {
 	_ = m.Write(&iom)
 	return int(iom.Histogram.GetSampleCount())
 }
+
+// GaugeValueWithLabels collects 10 samples with the provided labels from the
```

Comment:
I don't think this code actually collects 10 samples. It creates a channel of capacity 10, then does a single collection, right?

Reply:
You're correct :+1: